### PR TITLE
[PERF] on purchase confirm and low level domain_get

### LINKED ISSUE #25912: Performance issue on PO confirmation as UID!=1
**Impacted versions:**
v10, arround 4252cd70536 with some adaptions
I checked the code pieces and I deduce that later code bases, including v11 are also affected.

**Problem:**

- I duplicate PO A. After validating as `UID=1` it returns in about 1 sec
- I duplicate PO A. After validating as `UID!=1` it returns in about **70 sec**

**Analysis:**

The listed zips contain interactive tracing graphs for the different situations.
Teaser image: http://nimb.ws/Xb2thG

- [profiling.zip](https://github.com/odoo/odoo/files/2216530/profiling.zip) `UID!=1`
- [profiling2.zip](https://github.com/odoo/odoo/files/2216552/profiling2.zip) `UID==1`
- [profiling3.zip](https://github.com/odoo/odoo/files/2229502/profiling3.zip) A single notorious `stock_move.py create` with `UID!=1`

See also: https://github.com/odoo/odoo/issues/22002#issuecomment-406824232
OPW: 1868698

### PR DESCRIPTION
**Pull request by blaggacao**
Fixes #25912 
Maybe: #26001

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase/models/purchase.py [hunk 1]
```diff
@@ -717,11 +717,12 @@ def _prepare_stock_moves(self, picking):
 
     @api.multi
     def _create_stock_moves(self, picking):
-        moves = self.env['stock.move']
+        moves = self.env['stock.move'].with_context({'recompute': False})
```
Comment (gdgellatly) [inline addons/purchase/models/purchase.py]: I'm pretty sure this is a bad idea.  I think this completely overwrites the context, a lesson I've learnt the hard way.  What if there are settings in there like timezone, lang or date that are needed.  Better would be (notwithstanding whether the PR is appropriate)

`moves = self.env['stock.move'].with_context(recompute=False)`

Comment (blaggacao) [reply addons/purchase/models/purchase.py]: Thanks. Indeed. I changed that...

**Comment by blaggacao** [comment]
@nseinlet After https://github.com/odoo/odoo/issues/26001#issuecomment-408880152, still, the deferring  of recompute in this unfortunate case might be of interest.

**Comment by blaggacao** [comment]
Replaced by https://github.com/odoo/odoo/pull/26333

**Pull request closed without merging**